src/main/java/gim/model/exercise/ExerciseKeys.java

```diff
@@ -0,0 +1,40 @@
+package gim.model.exercise;
+
+import java.util.ArrayList;
+
+/**
+ * A class that contains the names of Keys in the Exercise Hashmap.
+ */
+public class ExerciseKeys {
+
+    private ArrayList<String> keyArrayList;
+
+    /**
+     * Creates a {@code ExerciseKeys}.
+     */
+    public ExerciseKeys(ArrayList<String> arrL) {
+        if (arrL == null) {
+            throw new NullPointerException("No Observer list found!");
```

Comment:
The exception messages (or other strings that may be used in multiple places) can be declared as `static` and `final` to prevent potential mutations to it in the future.

Reply:
Will be changed to an assertion instead, thanks.